src/Microsoft.OpenApi.OData.Reader/Edm/ODataPathProvider.cs

```diff
@@ -708,7 +708,10 @@ private void CreateTypeCastPaths(ODataPath currentPath, OpenApiConvertSettings c
         /// </summary>
         private void RetrieveBoundOperationPaths(OpenApiConvertSettings convertSettings)
         {
-            foreach (var edmOperation in _model.GetAllElements().OfType<IEdmOperation>().Where(e => e.IsBound))
+            // process functions before actions
+            var edmFunctions = _model.GetAllElements().OfType<IEdmFunction>().Where(e => e.IsBound).ToList();
+            var edmActions = _model.GetAllElements().OfType<IEdmOperation>().Where(e => e.IsBound && e is not IEdmFunction).ToList();
```

Comment:
```suggestion
            var edmActions = _model.GetAllElements().OfType<IEdmAction>().Where(e => e.IsBound);
```

Could you not do that instead?

Reply:
Reverted